Native/Text/src/ARText.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { createElement, useEffect, useRef, useState } from "react";
+import { ARTextProps } from "../typings/ARTextProps";
+import { Style } from "mendix";
+import { MeshComponent, setAttributes } from "../../../Shared/ComponentParent/src/MeshComponent";
+import { useGizmoComponent } from "../../../Shared/ComponentParent/src/useGizmoComponent";
+import { Mesh, MeshBuilder, Scene, Texture, Vector3 } from "@babylonjs/core";
+import { retrieveImageFromNumber } from "../../../Shared/ComponentParent/src/retrieveImageFromNumber";
+import font from "../../../Shared/ComponentParent/utils/Open Sans_Regular.json";
+import earcut from "earcut";
+
+export function ARText(props: ARTextProps<Style>): React.ReactElement {
+    (window as any).earcut = earcut;
+    const { mxMaterialTexture } = props;
+    const [mesh, setMesh] = useState<Mesh>();
+    const [scene, setScene] = useState<Scene>();
+    const [texture, setTexture] = useState<Texture>();
+    const gizmoTransform = useGizmoComponent({
+        mesh: mesh,
+        draggingEnabled: props.mxDraggingEnabled.value ?? false,
+        pinchEnabled: props.mxPinchEnabled.value ?? false,
+        rotationEnabled: props.mxPinchRotationEnabled.value ?? false,
+        gizmoSize: Number(props.mxGizmoSize.value) ?? 0.1,
+        color: props.mxGizmoColor.value ?? "#ffffff"
+    });
+
+    const handleSceneLoaded = (scene: Scene) => {
+        const newMesh = MeshBuilder.CreateText(
+            "myText",
+            props.mxText.value ?? "",
+            font,
+            {
+                resolution: 64,
+                depth: 10
+            },
+            scene
+        );
+        if (newMesh) {
+            newMesh.scaling = Vector3.Zero();
+            setMesh(newMesh);
+        }
+        setScene(scene);
+    };
+
+    useEffect(() => {
+        if (gizmoTransform.newPosition) {
+            setAttributes(gizmoTransform.newPosition, props.mxPositionXAtt, props.mxPositionYAtt, props.mxPositionZAtt);
+        }
+    }, [gizmoTransform.newPosition]);
+
+    useEffect(() => {
+        if (gizmoTransform.newRotation) {
+            setAttributes(gizmoTransform.newRotation, props.mxRotationXAtt, props.mxRotationYAtt, props.mxRotationZAtt);
+        }
+    }, [gizmoTransform.newRotation]);
+
+    useEffect(() => {
+        if (gizmoTransform.newScale) {
+            setAttributes(gizmoTransform.newScale, props.mxScaleXAtt, props.mxScaleYAtt, props.mxScaleZAtt);
+        }
+    }, [gizmoTransform.newScale]);
```

Comment:
NIT: you can merge all three hooks in one `useEffect`,

Reply:
Good point! I'll do that